bundles/org.openhab.binding.smhi/src/main/resources/ESH-INF/thing/channel-types.xml

```diff
@@ -0,0 +1,214 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="smhi"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-type id="msl">
+		<item-type>Number</item-type>
+		<label>Air Pressure</label>
+		<description>Air pressure in hPa</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="t">
+		<item-type>Number</item-type>
+		<label>Temperature</label>
+		<description>Temperature in C</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="vis" advanced="true">
+		<item-type>Number</item-type>
+		<label>Visibility</label>
+		<description>Horizontal visibility in km</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="wd">
+		<item-type>Number</item-type>
+		<label>Wind Direction</label>
+		<description>Wind direction in degrees</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="ws">
+		<item-type>Number</item-type>
+		<label>Wind Speed</label>
+		<description>Wind speed in m/s</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="r">
+		<item-type>Number</item-type>
+		<label>Relative Humidity</label>
+		<description>Relative humidity in percent</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="tstm" advanced="true">
+		<item-type>Number</item-type>
+		<label>Thunder Probability</label>
+		<description>Probability of thunder in percent</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="tcc_mean">
+		<item-type>Number</item-type>
+		<label>Total Cloud Cover</label>
+		<description>Mean value of total cloud cover in octets</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="lcc_mean" advanced="true">
+		<item-type>Number</item-type>
+		<label>Low Level Cloud Cover</label>
+		<description>Mean value of low level cloud cover (0-2500 m) in octets</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="mcc_mean" advanced="true">
+		<item-type>Number</item-type>
+		<label>Medium Level Cloud Cover</label>
+		<description>Mean value of medium level cloud cover (2500-6000 m) in octets</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="hcc_mean" advanced="true">
+		<item-type>Number</item-type>
+		<label>High Level Cloud Cover</label>
+		<description>Mean value of high level cloud cover (> 6000 m) in octets</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="gust">
+		<item-type>Number</item-type>
```

Comment:
Please apply openHab's [Units of Measurement](https://www.openhab.org/docs/concepts/units-of-measurement.html) framework wherever applicable. This allows user interfaces greater flexibility when it comes to rendering data from your binding.

```suggestion
		<item-type>Number:Speed</item-type>
```

Reply:
Thanks for your comments, working on implementing UoM. Do you now what's the best practice to handle precipitation? It's commonly reported as mm/h, but in reality it should be l/m²/h. It feels kind of weird to define it as a speed measurement.

Reply:
I would just stick with mm/h since it is more commonly used